### PR TITLE
Feature/11147 implement dynamic datatype casting

### PR DESCRIPTION
**Pull request by dennismeister93**
## Describe your changes

Implement dynamic datatype casting for datapoints of the vehicle databroker response

## Issue ticket number and link

#AB[11147](https://softwaredefinedvehicle.visualstudio.com/SoftwareDefinedVehicle/_workitems/edit/11147)

## Checklist - Manual tasks

* [x] Examples are executing successfully
* [x] Created/updated unit tests. Code Coverage percentage on new code shall be >= 80%.
* [x] Created/updated integration tests.
* [x] Devcontainer can be opened successfully
* [x] Devcontainer can be opened successfully behind a corporate proxy
* [x] Devcontainer can be re-built successfully
* [x] Extended the documentation (e.g. README.md, CONTRIBUTING.md, Velocitas)

### CODE REVIEW

File: sdv/vdb/reply.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+from sdv.proto.types_pb2 import Datapoint as BrokerDatapoint
+
+
+class DataPointReply:
+    """Wrapper for dynamic datatype casting of VDB reply."""
+
+    def __init__(self, reply):
+        self._reply = reply
+
+    def get(self, datapoint_object):
```
Comment (emsameen) [inline sdv/vdb/reply.py:24]: rename the public API name to datapoint

File: sdv/vdb/reply.py [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+        datapoint_type = datapoint_object.__class__.__name__
+        datapoint: BrokerDatapoint = self._reply.fields[datapoint_object.get_path()]
```
Comment (emsameen) [inline sdv/vdb/reply.py:26]: rename to vdm_datapoint

**Review by emsameen — CHANGES_REQUESTED** [review]
Rebase on main and Adjust the Dogmode app as well.

**Review by mml5bg — APPROVED** [review]

**Review by emsameen — APPROVED** [review]

**Review by emsameen — APPROVED** [review]
The remaining open points are updated and approved by other commiter.

**Pull request merged**